### PR TITLE
add function to define path

### PR DESCRIPTION
**Pull request by hyuraku**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?
- [ ] Have you successfully run `brew man` locally and committed any changes?

-----
refactor how to assign `only` whether `:formula` or `:cask` in commands like `cmd/fetch.rb`.

### CODE REVIEW

**Review by MikeMcQuaid — COMMENTED** [review]
Nice refactor! Perhaps in cases where `only` is used once then `args.only_path_formula_or_cask` could be passed as a parameter directly?

**Comment by hyuraku** [comment]
> Perhaps in cases where `only` is used once then `args.only_path_formula_or_cask` could be passed as a parameter directly?

Thank you for the advice. I will change `only` into `args.only_path_formula_or_cask`

File: Library/Homebrew/cli/args.rb [hunk 1]
```diff
@@ -133,6 +133,11 @@ def context
         Context::ContextStruct.new(debug: debug?, quiet: quiet?, verbose: verbose?)
       end
 
+      def only_path_formula_or_cask
```
Comment (reitermarkus) [inline Library/Homebrew/cli/args.rb:136]:
```suggestion
      def only_formula_or_cask
```

Not sure why `path` should be in the name.

**Comment by reitermarkus** [comment]
Given that all of these are passed from `Args` to `NamedArgs` which already contains a reference to `Args` (`self.parent`), we could just set the default value on the methods themselves, e.g.

```ruby
def to_paths(only: self.parent&.only_formula_or_path)
```

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks again @hyuraku!

**Pull request merged**